clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1469,7 +1470,19 @@ public static <K, V> Map<K, V> filterMap(final Map<K, V> map, final Predicate<En
      * @return a map including all elements in properties
      */
     public static Map<String, Object> propsToMap(Properties properties) {
-        return castToStringObjectMap(properties);
+        try {
+            final Enumeration<?> enumeration = properties.propertyNames();
+            Map<String, Object> props = new HashMap<>();
+            while (enumeration.hasMoreElements()) {
+                Object key = enumeration.nextElement();
+                String keyString = (String) key;
+                Object value = (properties.get(keyString) != null) ? properties.get(keyString) : properties.getProperty(keyString);
+                props.put(keyString, value);
+            }
+            return props;
+        } catch (Exception e) {
+            throw new ConfigException("Key must be a string.");
+        }
```

Comment:
Rather than casting the key to a `String`, my preference would be to proactively check the type, just like `castToStringObjectMap()` does:

```suggestion
        final Enumeration<?> enumeration = properties.propertyNames();
        Map<String, Object> map = new HashMap<>();
        while (enumeration.hasMoreElements()) {
            Object key = enumeration.nextElement();

            if (key instanceof String) {
                String k = (String) key;
                map.put(k, properties.getProperty(k));
            } else {
                throw new ConfigException(String.valueOf(key), properties.get(key), "Key must be a string.");
            }
        }
        return props;
```

Reply:
Hi Kirk, I attempted to implement `propsToMap` in the way you suggested above, but if we pass in a `Properties` object with non-string keys, a `ClassCastException` is thrown by `properties.propertyNames()`. My implementation above is the workaround I found. 

If it's assumed that all keys of the `Properties` object are `String`s, then I believe the code can be shortened, but might lead to some confusion during usage of `propsToMap`.

Reply:
In any case, let's also add the unit test for the exception thrown.